ext/standard/basic_functions.c

```diff
@@ -544,7 +544,8 @@ PHP_FUNCTION(inet_ntop)
 	char *address;
 	size_t address_len;
 	int af = AF_INET;
-	char buffer[40];
+	socklen_t buffersize = INET_ADDRSTRLEN;
+	char buffer[INET6_ADDRSTRLEN];
```

Comment:
Is anything `INET6_*` defined when `HAVE_IPV6` is not?

Also, is it safe to use shorter `buffersize` when compiled /wo IPv6 support and then run on system /w IPv6 support?

Reply:
> Is anything `INET6_*` defined when `HAVE_IPV6` is not?

Yes it is independant from ipv6 support and, alone, does not indicate the system supports it.
> 
> Also, is it safe to use shorter `buffersize` when compiled /wo IPv6 support and then run on system /w IPv6 support?

Yes in that case the best course is to reverse the logic.